compiler/rustc_transmute/src/maybe_transmutable/tests.rs

```diff
@@ -1,93 +1,115 @@
 use itertools::Itertools;
 
 use super::query_context::test::{Def, UltraMinimal};
-use crate::maybe_transmutable::MaybeTransmutableQuery;
-use crate::{Reason, layout};
+use crate::{Answer, Assume, Reason, layout};
 
-mod safety {
-    use super::*;
-    use crate::Answer;
+type Tree = layout::Tree<Def, !>;
+type Dfa = layout::Dfa<!>;
 
-    type Tree = layout::Tree<Def, !>;
+trait Representation {
+    fn is_transmutable(src: Self, dst: Self, assume: Assume) -> Answer<!>;
+}
```

Comment:
It took me a moment to understand the purpose of this trait, but it's cute! So cute, in fact, that I think it would be informative to have the `Representation` bound on `MaybeTransmutableQuery`. 

Could you move `Representation` to the `layout` module, remove its method, and then bound the `L` parameter of `MaybeTransmutableQuery` with `Representation`? This has the benefit of making `MaybeTransmutableQuery` more self-documenting, and the benefit of controlling the proliferation of ways to ask "is this transmutable".

Reply:
Discussed offline: Turns out this refactoring is not as straightforward as suggested, and so we're going to stick with it as-is.

Reply:
Discussed offline; this is a bigger refactor than anticipated. The PR is fine as-is; we can follow up.